src/main/java/com/bankingservice/banking/exception/ApplicationExceptionHandler.java

```diff
@@ -0,0 +1,43 @@
+package com.bankingservice.banking.exception;
+
+import com.bankingservice.banking.dto.response.BaseResponseDTO;
+import com.bankingservice.banking.dto.response.MetaDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class ApplicationExceptionHandler {
+    private static final Logger logger = LoggerFactory.getLogger(ApplicationExceptionHandler.class);
+
+    /**
+     * Handles insertion failed exception
+     *
+     * @param e
+     * @return response entity
+     */
+    @ExceptionHandler(InsertionFailedException.class)
+    public ResponseEntity<BaseResponseDTO> handleInsertionFailedException(InsertionFailedException e){
+        logger.error("InsertionFailedException has occurred!");
```

Comment:
In exception error log add the stack trace of the exception caused is RuntimeExcetion and custom for logical errors.